packages/internals/support/src/entities/component.js

```diff
@@ -64,6 +51,25 @@ class Component extends Entity {
       get: variantsGet,
       set: variantsSet
     });
+
+    Object.defineProperty(this, 'config', {
+      enumerable: true,
+      get: configGet,
+      set: configSet
+    });
+  }
+
+  get(path, fallback) {
+    fallback = get(this.config, path, fallback);
+    const result = get(_data.get(this), path, fallback);
+    return isObjectLike(result) ? cloneDeep(result) : result;
+  }
+
+  set(path, value) {
+    assert.string(path, 'Component.set - `path` argument must be a string [path-invalid]');
+    // TODO: extract shared caching code with config: this.removeFromCache(path);
+    set(_data.get(this), path, cloneDeep(value));
```

Comment:
Have you checked this with non-array/object values? I think i you try and do cloneDeep on strings it might behave oddly (like return an array) which is why i used `isObjectLike(result) ? cloneDeep(result) : result;` in the getter... may have been me just being an idiot but worth double checking.